.github/workflows/run-tests.yml

```
@@ -0,0 +1,75 @@
+      
+      - name: Final Check
+        id: final_check
+        if: always() # Ensure this step always runs
+        run: |
+          if [[ "${{ steps.execute_all_tests.outcome }}" == "success" ]]; then
+            echo "All tests passed."
+            exit 0
+          else
+            echo "One or more tests failed."
+            exit 1
+          fi 
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)